### PR TITLE
[ADD] Update developer documentation

### PR DESCRIPTION
**Pull request by ravinkohli**

### CODE REVIEW

**Review by nabenabe0928 — CHANGES_REQUESTED** [review]
Hi thanks for the PR and I pushed suggestions via a separated file
Please check and modify the file.

File: docs/dev_suggestion.rst [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+#. PyTorch neural networks: Neural architecture search of backbones (feature extraction) and network heads (for the final prediction)
+
+Note that dummy pipeline is used as a baseline to define the worst-performing model that can be fit
+and traditional machine learning pipelines are critical for small-sized datasets.
+A pipeline might also contain additional training components
+like learning rate scheduler, optimizers,
+and data loaders required to perform the neural architecture search.
+
+After the training (fitting a pipeline), we use pickle to save it
+to disk as stated `here <https://scikit-learn.org/stable/modules/model_persistence.html>`_.
```
Comment (ravinkohli) [inline docs/dev_suggestion.rst:79]:
```suggestion
to disk as stated `here <https://scikit-learn.org/stable/modules/model_persistence.html>`_. Additionally, we save the predictions of the pipeline on the test set and the optimisation set in a `numpy NPY format <https://numpy.org/doc/stable/reference/generated/numpy.lib.format.html#module-numpy.lib.format>`_
```

File: docs/dev_suggestion.rst [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+Optimization of pipeline
+------------------------
+
+To optimize pipeline, we use SMAC as mentioned earlier.
```
Comment (ravinkohli) [inline docs/dev_suggestion.rst:84]:
```suggestion
To optimize the pipeline, we use SMAC as mentioned earlier.
```

File: docs/dev_suggestion.rst [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+Given a configuration, AutoPytorch pipeline fits a pipeline and
```
Comment (ravinkohli) [inline docs/dev_suggestion.rst:85]:
```suggestion
Given a configuration, AutoPytorch fits a pipeline and 
```

File: docs/dev_suggestion.rst [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+After the training (fitting a pipeline), we use pickle to save it
+to disk as stated `here <https://scikit-learn.org/stable/modules/model_persistence.html>`_.
+
+Optimization of pipeline
+------------------------
+
+To optimize pipeline, we use SMAC as mentioned earlier.
+Given a configuration, AutoPytorch pipeline fits a pipeline and 
+and finally saves to disc using the function evaluator `ExecuteTaFuncWithQueue`.
```
Comment (ravinkohli) [inline docs/dev_suggestion.rst:86]:
```suggestion
 finally saves to disc using the function evaluator `ExecuteTaFuncWithQueue`.
```

File: docs/dev.rst [hunk 5]
```diff
@@ -32,29 +48,94 @@ These steps include-
...
+
+After the training (fitting a pipeline), we use pickle to save it
+to disk as stated `here <https://scikit-learn.org/stable/modules/model_persistence.html>`_.
+
+Optimization of pipeline
+------------------------
+
+To optimize the pipeline, we use SMAC as mentioned earlier.
+Given a configuration, AutoPytorch fits a pipeline and 
+ finally saves to disc using the function evaluator `ExecuteTaFuncWithQueue`.
```
Comment (nabenabe0928) [inline docs/dev.rst:86]: Remove space before `finally` (it seems it makes this line a quotation)

**Review by nabenabe0928 — CHANGES_REQUESTED** [review]

File: docs/dev.rst (hunk 5 above)
Comment (ravinkohli) [reply docs/dev.rst:86]:
```suggestion
finally saves to disc using the function evaluator `ExecuteTaFuncWithQueue`.
```

**Review by nabenabe0928 — APPROVED** [review]

**Comment by nabenabe0928** [comment]
Hi, we can merge this PR.

**Pull request merged**